sql/ddl_internal.sql

```diff
@@ -259,26 +272,39 @@ $BODY$
 $BODY$;
 
 -- Drop chunks older than the given timestamp. If a hypertable name is given,
--- drop only chunks associated with this table.
+-- drop only chunks associated with this table. Any of the three arguments
+-- can be NULL meaning "all values".
 CREATE OR REPLACE FUNCTION _timescaledb_internal.drop_chunks_older_than(
     older_than_time  BIGINT,
     table_name  NAME = NULL,
     schema_name NAME = NULL
 )
     RETURNS VOID LANGUAGE PLPGSQL VOLATILE AS
 $BODY$
+DECLARE 
+    chunk_row _timescaledb_catalog.chunk;
 BEGIN
-    PERFORM _timescaledb_internal.drop_chunk(c.id, false)
-    FROM _timescaledb_catalog.chunk c
-    INNER JOIN _timescaledb_catalog.hypertable h ON (h.id = c.hypertable_id)
-    INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
-    INNER JOIN _timescaledb_catalog.dimension_slice ds
-        ON (ds.dimension_id =  time_dimension.id)
-    INNER JOIN _timescaledb_catalog.chunk_constraint cc
-        ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
-    WHERE ds.range_end <= older_than_time
-    AND (drop_chunks_older_than.schema_name IS NULL OR h.schema_name = drop_chunks_older_than.schema_name)
-    AND (drop_chunks_older_than.table_name IS NULL OR h.table_name = drop_chunks_older_than.table_name);
+    IF older_than_time IS NULL AND table_name IS NULL AND schema_name IS NULL THEN
+        RAISE 'Cannot have all 3 arguments to drop_chunks_older_than be NULL';
+    END IF;
+    FOR chunk_row IN SELECT *
+        FROM _timescaledb_catalog.chunk c
+        INNER JOIN _timescaledb_catalog.hypertable h ON (h.id = c.hypertable_id)
+        INNER JOIN _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
+        INNER JOIN _timescaledb_catalog.dimension_slice ds
+            ON (ds.dimension_id = time_dimension.id)
+        INNER JOIN _timescaledb_catalog.chunk_constraint cc
+            ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
+        WHERE (older_than_time IS NULL OR ds.range_end <= older_than_time)
+        AND (drop_chunks_older_than.schema_name IS NULL OR h.schema_name = drop_chunks_older_than.schema_name)
+        AND (drop_chunks_older_than.table_name IS NULL OR h.table_name = drop_chunks_older_than.table_name)
+    LOOP
+        EXECUTE format(
+                $$
+                DROP TABLE %I.%I
+                $$, chunk_row.schema_name, chunk_row.table_name
```

Comment:
If we don't have CASCADE support here, what happens when there are dependent objects and someone calls `drop_chunks_older_than()` as part of a retention policy script? I guess the retention policy will simply fail and the database might, e.g., run out of disk space? That sounds bad. CASCADE should at least be an option IMO.

Reply:
Yep  adding an option is probably a good idea. Will do.